monai/handlers/roc_auc.py

```diff
@@ -96,6 +98,17 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
     def compute(self):
         _prediction_tensor = torch.cat(self._predictions, dim=0)
         _target_tensor = torch.cat(self._targets, dim=0)
+
+        if dist.is_available() and dist.is_initialized() and not self._is_reduced:
+            # create placeholder to collect the data from all processes:
+            output = [torch.zeros_like(_prediction_tensor) for _ in range(dist.get_world_size())]
+            dist.all_gather(output, _prediction_tensor)
```

Comment:
Here it may fail if `_prediction_tensor` is not on the correct device according to dist backend. For example, nccl <-> "cuda:local_rank" or gloo <-> "cpu"

Reply:
I see, thanks for your reminder.